core/src/services/routing.js

```diff
@@ -662,18 +665,21 @@ class RoutingClass {
       } else {
         url.search = `?${RoutingHelpers.encodeParams(params)}`;
       }
-      history.replaceState(window.state, '', url.href);
+      if (!sessionStorage.getItem('historyState')) {
+        sessionStorage.setItem('historyState', history.length);
+      }
+      history.pushState(window.state, '', url.href);
     }
   }
 
-  removeModalDataFromUrl() {
+  removeModalDataFromUrl(closeBtnPressed) {
```

Comment:
Could you please add a JSDOC description on this function

Reply:
👍